control-plane/api/v2alpha1/traffic_permissions_types_test.go

```diff
@@ -0,0 +1,811 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v2alpha1
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v1alpha1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/testing/protocmp"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/hashicorp/consul-k8s/control-plane/api/common"
+	inject "github.com/hashicorp/consul-k8s/control-plane/connect-inject/common"
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	"github.com/hashicorp/consul-k8s/control-plane/helper/test"
+)
+
+func TestTrafficPermissions_MatchesConsul(t *testing.T) {
+	cases := map[string]struct {
+		OurConsulNamespace string
+		OurConsulPartition string
+		OurData            TrafficPermissions
+
+		TheirName            string
+		TheirConsulNamespace string
+		TheirConsulPartition string
```

Comment:
Wow, thanks for adding ent here, I don't think the other config entries had ent tests like this. So thorough!!